src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -530,10 +533,10 @@ internal void ProcessPendingAck(TdsParserStateObject stateObj)
             if (connHandler.ConnectionOptions.LocalDBInstance != null)
             {
                 LocalDBAPI.CreateLocalDBInstance(connHandler.ConnectionOptions.LocalDBInstance);
-                if (encrypt)
+                if (encrypt == SqlConnectionEncryptOption.Mandatory)
```

Comment:
Same comment as on the netcore version. We only want to automatically change to optional if encrypt=true/mandatory because it likely means they are using the new default value. If they have set encrypt=strict, they have explicitly set it and localdb connections will fail since they don't support encryption.